Rakefile

```diff
@@ -56,6 +57,41 @@ def format_links(text)
   text.gsub(/{([^}]+)}/, '[`\1`](#\1)')
 end
 
+desc "Generate markdown docs for Bolt's command line options"
+task :reference_docs do
```

Comment:
In order to keep the rake file organized I think we should make a docs namespace similar to the integration namespace. Then create a task that runs everything under that namespace for use in travis or building docs locally to test out changes. 
So maybe a namespace called `docs`, then nest two tasks `function_reference` and `cli_reference` under those, you could make a task like
```
desc "generate docs"
task generate_docs: [:docs, 'plan_function', 'cli_reference']
```